subsys/mgmt/updatehub/updatehub.c

```diff
@@ -946,6 +953,8 @@ enum updatehub_response updatehub_update(void)
 
 	LOG_INF("Image flashed successfully, you can reboot now");
 
+	k_msleep(2000);
```

Comment:
Why are you sleeping here?

Reply:
When in polling mode need flush logs. Otherwise there is no "success" indicator after reboot. This keeps docs consistent with what users see.

Reply:
Couldn't UpdateHub use the immediate mode? As this is critical, it seems fine as fas we have a minimal number of log messages.

Reply:
I do not recommend to set `CONFIG_LOG_IMMEDIATE` as it can slow down things if there is lot of logging output.
Just wondering why it is important that the user sees the info message? It could be that the info messages are not printed or there is no console where to print these. Shouldn't the status code be propagated up in the stack and eventually to application etc?

Reply:
I agreed with Jukkar, CONFIG_LOG_IMMEDIATE is not the solution. The message isn't print because on polling mode after a successful upgrade system reboots immediately and buffered messages don't have time to be printed. Ideally a "LOG_FLUSH" is enough but I'm not sure if there is this kind of think available.

> Shouldn't the status code be propagated up in the stack and eventually to application etc?

In pooling mode UpdateHub takes control and reboot without communicate upper layers. When user wants control, it should execute **Manually** and check returned status. Polling mode is for simple apps like sensors that an eventual reset doesn't compromise system.

Reply:
I wonder if `log_panic()` could be used here. Its documentation says that the logs are flushed. And as we reboot the device immediately, it should not matter if we enter the panic mode anyway.
```
/**
 * @brief Switch the logger subsystem to the panic mode.
 *
 * Returns immediately if the logger is already in the panic mode.
 *
 * @details On panic the logger subsystem informs all backends about panic mode.
 *          Backends must switch to blocking mode or halt. All pending logs
 *          are flushed after switching to panic mode. In panic mode, all log
 *          messages must be processed in the context of the call.
 */
__syscall void log_panic(void);
```

Reply:
Oh ... the panic kinda makes sense as we'll be forcing flushing and also informing other subsystems it'll reboot does sounds right.

Reply:
I added LOG_PANIC() and tested with frdm_k64f.